cpp/src_prims/selection/knn.cuh

```diff
@@ -174,8 +173,8 @@ template <int TPB_X = 32, bool precomp_lbls = false>
 void class_probs(std::vector<float*>& out,
                  const int64_t* knn_indices,
                  std::vector<int*>& y,
-                 size_t n_index_rows,
-                 size_t n_query_rows,
+                 int n_index_rows,
```

Comment:
It's not immediately clear to me why these need to be `int`s. Don't think you're wrong, but can you tell me what the thinking is here?

Reply:
Just adding a small comment here that many of the arguments / variables that store matrix size info are intentionally `size_t` because we've been bitten by overflows in the past.

Reply:
I changed it because the only place knn_classify is used is in tests, and those tests pass int.  I will revert this change since now I see this is a public API, my mistake.

Reply:
Fixed.